errors/error.go

```diff
@@ -94,6 +82,31 @@ func New(ctx context.Context, args ...interface{}) error {
 	return e
 }
 
+// WithContext sets the context of the error. However, it does not override the existing context if it has already been
+// set. In the cases where there is a child node, the child node's context would have already been propagated to the
+// current context.
+func (e *Error) WithContext(ctx context.Context) *Error {
+	if e.ctx == nil {
+		e.ctx = map[string]any{}
+	}
+
+	basics := common.GetBasics(ctx)
+	if basics != nil {
+		currentBasics, _ := e.ctx[ctxBasics].(common.Basics)
```

Comment:
Can we use this?
I find it easier to read.
```suggestion
		currentBasics, _ := e.basics()
```

Reply:
Agreed. Refactored in 49cae4f. Meanwhile, I also added setter methods to make it easier to read, too.